File: swagger-client/csharp-dotnet2/src/main/CsharpDotNet2/mailbaby-client-csharp-dotnet2/Model/MailLog.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;
using System.Runtime.Serialization;
using Newtonsoft.Json;

namespace IO.Swagger.Model {

  /// <summary>
  /// Paginated mail log response returned by &#x60;GET /mail/log&#x60;.  Contains the full matched count (&#x60;total&#x60;) plus a page of &#x60;MailLogEntry&#x60; records.  Use &#x60;skip&#x60; and &#x60;limit&#x60; to page through large result sets.
  /// </summary>
  [DataContract]
  public class MailLog {
    /// <summary>
    /// Total number of log entries that match the supplied filters, regardless of `skip` and `limit`.  Use this to calculate the number of pages: `ceil(total / limit)`.
    /// </summary>
    /// <value>Total number of log entries that match the supplied filters, regardless of `skip` and `limit`.  Use this to calculate the number of pages: `ceil(total / limit)`.</value>
    [DataMember(Name="total", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "total")]
    public int? Total { get; set; }

    /// <summary>
    /// The `skip` value used for this page (echoed from the request).
    /// </summary>
    /// <value>The `skip` value used for this page (echoed from the request).</value>
    [DataMember(Name="skip", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "skip")]
    public int? Skip { get; set; }

    /// <summary>
    /// The `limit` value used for this page (echoed from the request).
    /// </summary>
    /// <value>The `limit` value used for this page (echoed from the request).</value>
    [DataMember(Name="limit", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "limit")]
    public int? Limit { get; set; }

    /// <summary>
    /// Gets or Sets Emails
    /// </summary>
    [DataMember(Name="emails", EmitDefaultValue=false)]
    [JsonProperty(PropertyName = "emails")]
    public List<MailLogEntry> Emails { get; set; }


    /// <summary>
    /// Get the string presentation of the object
    /// </summary>
    /// <returns>String presentation of the object</returns>
    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class MailLog {\n");
      sb.Append("  Total: ").Append(Total).Append("\n");
      sb.Append("  Skip: ").Append(Skip).Append("\n");
      sb.Append("  Limit: ").Append(Limit).Append("\n");
      sb.Append("  Emails: ").Append(Emails).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }

    /// <summary>
    /// Get the JSON string presentation of the object
    /// </summary>
    /// <returns>JSON string presentation of the object</returns>
    public string ToJson() {
      return JsonConvert.SerializeObject(this, Formatting.Indented);
    }

}
}

File: openapi-client/javascript-closure-angular/API/Client/MailLog.js
goog.provide('API.Client.MailLog');

/**
 * Paginated mail log response returned by `GET /mail/log`.  Contains the full matched count (`total`) plus a page of `MailLogEntry` records.  Use `skip` and `limit` to page through large result sets.
 * @record
 */
API.Client.MailLog = function() {}

/**
 * Total number of log entries that match the supplied filters, regardless of `skip` and `limit`.  Use this to calculate the number of pages: `ceil(total / limit)`.
 * @type {!number}
 * @export
 */
API.Client.MailLog.prototype.total;

/**
 * The `skip` value used for this page (echoed from the request).
 * @type {!number}
 * @export
 */
API.Client.MailLog.prototype.skip;

/**
 * The `limit` value used for this page (echoed from the request).
 * @type {!number}
 * @export
 */
API.Client.MailLog.prototype.limit;

/**
 * @type {!Array<!API.Client.MailLogEntry>}
 * @export
 */
API.Client.MailLog.prototype.emails;

